File: Classes/EVEOnlineAPI/Server/EVEServerStatus.m
//
//  EVEServerStatus.m
//  EVEOnlineAPI
//
//  Created by Artem Shimanski on 8/20/10.
//  Copyright 2010 __MyCompanyName__. All rights reserved.
//

#import "EVEServerStatus.h"


@implementation EVEServerStatus

+ (NSDictionary*) scheme {
	static NSDictionary* scheme = nil;
	if (!scheme)
		scheme = @{@"serverOpen":@{@"type":@(EVEXMLSchemePropertyTypeScalar), @"transformer":^(id value) {
			if ([value isKindOfClass:[NSString class]])
				return @([value boolValue]);
			else
				return @(NO);
		}},
				   @"onlinePlayers":@{@"type":@(EVEXMLSchemePropertyTypeScalar)}};
	
	return scheme;
}

@end

File: EVEAPI/EVEAPI/EVEServerStatus.swift
//
//  EVEServerStatus.swift
//  EVEAPI
//
//  Created by Artem Shimanski on 28.11.16.
//  Copyright © 2016 Artem Shimanski. All rights reserved.
//

import UIKit

public class EVEServerStatus: EVEResult {
	public var serverOpen: Bool = false
	public var onlinePlayers: Int = 0
	
	public required init?(dictionary:[String:Any]) {
		super.init(dictionary: dictionary)
	}
	
	public required init?(coder aDecoder: NSCoder) {
		super.init(coder: aDecoder)
	}

	override public func scheme() -> [String:EVESchemeElementType] {
		return [
			"serverOpen":EVESchemeElementType.Bool(elementName:nil, transformer:nil),
			"onlinePlayers":EVESchemeElementType.Int(elementName: nil, transformer: nil)
		]
	}
	
}
